### PR TITLE
Fix transforms passed to each dataset split

### PR DESCRIPTION
**Pull request by sfmig**
We were passing the transforms of the training set to all other sets.

With this fix we pass the test transforms to the validation set and the test set.

Feels a bit overkill to compute all three splits every time, but to use `random_split` we need to pass a dataset with transforms already linked to it. Open to suggestions for a nicer way 👀 .

Also: how could we make a test for this?

### CODE REVIEW

File: crabs/detection_tracking/datamodules.py [hunk 1]
```diff
@@ -163,18 +166,27 @@ def setup(self, stage: str):
...
         self.train_transform = self._get_train_transform()
         self.test_transform = self._get_test_val_transform()
         self.val_transform = self._get_test_val_transform()
 
-        # Assign datasets for dataloader depending on stage
+        # Assign datasets for dataloader depending on stage.
         # omitting "predict" stage for now
-        train_dataset, test_dataset, val_dataset = self._compute_splits()
         if stage == "fit":
+            # compute splits passing the relevant transform
```
Comment (nikk-nikaznan) [inline crabs/detection_tracking/datamodules.py:177]: we don't need this condition actually. Lightning will deal with it.
after assigning the transform..
```
self.train_dataset, _, _ = self._compute_splits(self.train_transform)
_, _, self.val_dataset = self._compute_splits(self.val_transform)
_, self.test_dataset, _ = self._compute_splits(self.test_transform)
```

File: crabs/detection_tracking/datamodules.py [hunk 2]
```diff
@@ -117,7 +120,7 @@ def _compute_splits(
         full_dataset = CrabsCocoDetection(
             self.list_img_dirs,
             self.list_annotation_files,
-            transforms=self.train_transform,
+            transforms=transforms_all_splits,  # self.train_transform,
```
Comment (nikk-nikaznan) [inline crabs/detection_tracking/datamodules.py:123]: I overlooked this, so before this we apply train_transform to all the images?

**Review by nikk-nikaznan — APPROVED** [review]
I have tested on my end to remove the condition for stage and it works the same. From lighting latest documentation we don't really need that.

In term of the transform, I guess it's fine for now. I will try to dig more better to do this, but it's not making the loading much slower or anything

File: crabs/detection_tracking/datamodules.py (hunk 1 above)
Comment (sfmig) [reply crabs/detection_tracking/datamodules.py:177]: I couldn't find this update in the docs - would you mind linking to it? 

So without the stage we now compute all splits, regardless of the stage. Is that correct?
